### PR TITLE
Add an option to log all screenshots server side

### PR DESCRIPTION
**Pull request by renajohn**
This PR adds the option to dump all taken images on disk to a specified
folder.

### CODE REVIEW

File: java/server/src/org/openqa/selenium/server/RemoteControlConfiguration.java [hunk 1]
```diff
@@ -121,6 +123,15 @@ public void setProfilesLocation(File profilesLocation) {
     this.profilesLocation = profilesLocation;
   }
 
+  public File getDumpScreenshots() {
+    return dumpScreenshots;
+  }
+
+  public void setDumpScreenshots(File dumpScreenshots) {
```
Comment (kart) [inline java/server/src/org/openqa/selenium/server/RemoteControlConfiguration.java:130]: Can we name it something like : getScreenshotLocation and setScreenshotLocation? 'dump' doesn't sound right with screenshots..

File: java/server/src/org/openqa/selenium/remote/server/handler/CaptureScreenshot.java [hunk 2]
```diff
@@ -17,27 +17,71 @@
...
+  private File save(File folder, byte[] data) {
+    OutputStream stream = null;
+
+    try {
+      String filename = System.currentTimeMillis() + "_screenshot.png";
+
+      File file = new File(folder, filename);
+
+      stream = new FileOutputStream(file);
+      stream.write(data);
```
Comment (kart) [inline java/server/src/org/openqa/selenium/remote/server/handler/CaptureScreenshot.java:72]: I would suggest using an abstraction here: Guava Files API (Selenium already has the dependency wired in).
`Files.write(data, new File(folder, filename))`

This way, you can avoid headaches about closing the streams and such.

File: java/server/src/org/openqa/selenium/remote/server/handler/CaptureScreenshot.java [hunk 3]
```diff
@@ -17,27 +17,71 @@
...
+import java.io.OutputStream;
+
 public class CaptureScreenshot extends WebDriverHandler<String> {
 
+  private File dumpDir;
+
   public CaptureScreenshot(Session session) {
     super(session);
+
+    String path = System.getProperty("dumpScreenshots");
```
Comment (kart) [inline java/server/src/org/openqa/selenium/remote/server/handler/CaptureScreenshot.java:40]: Why do we need this if there is a command line parameter for setting the screenshot location?

Comment (renajohn) [inline java/server/src/org/openqa/selenium/remote/server/handler/CaptureScreenshot.java:40]: I couldn't find a way for the `CaptureScreenshot` to get access to the configuration object.

File: java/server/src/org/openqa/selenium/remote/server/handler/CaptureScreenshot.java (hunk 2 above)
Comment (renajohn) [inline java/server/src/org/openqa/selenium/remote/server/handler/CaptureScreenshot.java:72]: If Guava is already in Selenium, why not.

**Pull request closed without merging**